validate_test.go

```diff
@@ -64,6 +64,32 @@ func TestRequiredString(t *testing.T) {
 	}
 }
 
+func TestRequiredSlice(t *testing.T) {
+	tests := []struct {
+		a    interface{}
+		want bool
```

Comment:
Maybe change this to `haserrors` so it's a bit more clear.  Seeing the `true` in there makes me see the inverse.